### PR TITLE
[skip actions] NH-26618: update init msg for swo

### PR DESCRIPTION
**Pull request by xuan-cao-swi**
## Why?

To update the __init msg based on the [requirements](https://github.com/librato/trace/blob/master/docs/specs/KV/init.md#swo)

## Impact

Changed msg
```
{"__Init"=>true,                                                                                   
 "APM.Version"=>"5.1.4",                                                                           
 "APM.Extension.Version"=>"11.1.0",                                                                
 "process.pid"=>28128,                                                                             
 "process.command"=>"irb",                                                                         
 "process.runtime.name"=>"ruby",                                                                   
 "process.runtime.version"=>"2.7.5",                                                               
 "process.runtime.description"=>"ruby 2.7.5p203 (2021-11-24 revision f69aeb8314) [x86_64-linux]",  
 "telemetry.sdk.language"=>nil,                                                                    
 "process.executable.path"=>"/root/.rbenv/versions/2.7.5/bin/ruby",                                
 "process.executable.name"=>"ruby",                                                                
 "process.command_line"=>"irb",                                                                    
 "process.telemetry.path"=>"/root/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/solarwinds_apm-5.1.4",
 "os.type"=>"x86_64-linux",                                           
 "os.description"=>"Linux docker.ao.ubuntu 5.15.49-linuxkit #1 SMP PREEMPT Tue Sep 13 07:51:32 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux",
 "Ruby.did_you_mean.Version"=>"1.4.0",
 "Ruby.reline.Version"=>"0.3.2",
 "Ruby.irb.Version"=>"1.6.2",
 "Ruby.uri.Version"=>"0.10.0",
 "Ruby.io-console.Version"=>"0.6.0",
 "Ruby.timeout.Version"=>"0.1.0",
 "Ruby.forwardable.Version"=>"1.3.1",
 "Ruby.delegate.Version"=>"0.1.0",
 "Ruby.fileutils.Version"=>"1.4.1",
 "Ruby.etc.Version"=>"1.1.0",
 "Ruby.fiddle.Version"=>"1.0.0",
 "Ruby.rdoc.Version"=>"6.2.1.1",
 "Ruby.readline.Version"=>"0.0.2",
 "Ruby.readline-ext.Version"=>"0.1.0",
 "Ruby.date.Version"=>"3.0.3",
 "Ruby.cgi.Version"=>"0.1.0.1",
 "Ruby.strscan.Version"=>"1.0.3",
 "Ruby.no_proxy_fix.Version"=>"0.1.2",
 "Ruby.solarwinds_apm.Version"=>"5.1.4",
 "Ruby.ipaddr.Version"=>"1.2.2",
 "Ruby.openssl.Version"=>"2.1.3",
 "Ruby.stringio.Version"=>"0.1.0",
 "Ruby.logger.Version"=>"1.4.2",
 "Ruby.singleton.Version"=>"0.1.0",
 "Ruby.json.Version"=>"2.6.3",
 "Ruby.ostruct.Version"=>"0.2.0",
 "Ruby.zlib.Version"=>"1.1.0"}
```

### CODE REVIEW

File: lib/solarwinds_apm/util.rb [hunk 1]
```diff
@@ -265,57 +265,68 @@ def legacy_build_init_report
...
+        begin
+          platform_info['APM.Version']                  = SolarWindsAPM::Version::STRING
+          platform_info['APM.Extension.Version']        = get_extension_lib_version
+          
+          platform_info['process.pid']                  = Process.pid
+          platform_info['process.command']              = $PROGRAM_NAME
+          platform_info['process.runtime.name']         = RUBY_ENGINE
+          platform_info['process.runtime.version']      = RUBY_VERSION
+          platform_info['process.runtime.description']  = RUBY_DESCRIPTION
+          platform_info['telemetry.sdk.language']       = nil
```
Comment (cheempz) [inline lib/solarwinds_apm/util.rb:318]: `telemetry.sdk.language` is required by our spec and should be set as "ruby".

File: lib/solarwinds_apm/util.rb [hunk 2]
```diff
@@ -265,57 +265,68 @@ def legacy_build_init_report
...
+      # build_legacy_ao_init_report
       #
       # Internal: Build a hash of KVs that reports on the status of the
       # running environment.  This is used on stack boot in __Init reporting
       # and for SolarWindsAPM.support_report.
+      # On 2023-01-12, this is no longer used in new SWO endpoints
       #
       # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/AbcSize
-      def build_init_report
+      def build_legacy_ao_init_report
```
Comment (cheempz) [inline lib/solarwinds_apm/util.rb:276]: Minor, no harm keeping this but the idea is SWO Ruby will no longer report the legacy format.

**Review by cheempz — COMMENTED** [review]
Thanks @xuan-cao-swi!  I left a few comments.

File: lib/solarwinds_apm/util.rb (hunk 2 above)
Comment (xuan-cao-swi) [reply lib/solarwinds_apm/util.rb:276]: Yes, the build_legacy_ao_init_report will never be called.

File: test/support/init_report_test.rb [hunk 3]
```diff
@@ -12,7 +12,7 @@
    it 'report_kvs' do
     init_kvs = ::SolarWindsAPM::Util.build_init_report
     _(init_kvs.has_key?("__Init")).must_equal true
-    _(init_kvs.has_key?("Force")).must_equal true
+    _(init_kvs.has_key?("Force")).must_equal false # removed for swo
```
[the remark this answers is not in the document]
Comment (xuan-cao-swi) [reply test/support/init_report_test.rb]: Updated.

File: lib/solarwinds_apm/util.rb (hunk 1 above)
Comment (xuan-cao-swi) [reply lib/solarwinds_apm/util.rb]: fixed

**Review by cheempz — APPROVED** [review]
👍 thanks @xuan-cao-swi

**Pull request merged**